src/cli.rs

```diff
@@ -0,0 +1,136 @@
+//! Command-line argument parser and argument access.
+//!
+//! To access arguments at any point in the program, use [`with_opts`] or [`with_opts_mut`].
+
+use std::sync::Mutex;
+
+use camino::Utf8PathBuf;
+use clap::Parser;
+use once_cell::sync::OnceCell;
+
+/// The current command-line options.
+///
+/// Access these options with [`with_opts`] and [`with_opts_mut`].
+static OPTS: Mutex<OnceCell<Opts>> = Mutex::new(OnceCell::new());
```

Comment:
It needs to be in a `Mutex` because it's a `static` item (shared between all threads in the program), so we need to ensure that other threads aren't touching it before we access it. (There's a few other ways to do this, like with a [`RwLock`](https://doc.rust-lang.org/stable/std/sync/struct.RwLock.html) or `unsafe` code, but this is the easiest way.)

Reply:
The docs for `once_cell::sync::OnceCell` ([link](https://docs.rs/once_cell/latest/once_cell/sync/struct.OnceCell.html)) have an example with a `static` item being accessed from another thread without a `Mutex` wrapper. What's different about this case?

Reply:
`OnceCell` uses `unsafe` and an `AtomicPtr` to synchronize access to the contained item. I actually prototyped this using [`OnceLock`](https://doc.rust-lang.org/stable/std/sync/struct.OnceLock.html) (which is basically `OnceCell`), but it was only stabilized in Rust 1.70, which we don't have in the `mercury-web-backend` `nixpkgs` yet.

But also `OnceCell` doesn't provide mutable access on its own, because it's not safe to take an `&mut` reference to a `static` item:

```rust
use once_cell::sync::OnceCell;

static CELL: OnceCell<String> = OnceCell::new();

fn main() {
    CELL.get_mut();
}
```

```
error[E0596]: cannot borrow immutable static item `CELL` as mutable
 --> src/main.rs:6:5
  |
6 |     CELL.get_mut();
  |     ^^^^^^^^^^^^^^ cannot borrow as mutable
```

[E0596](https://doc.rust-lang.org/stable/error_codes/E0596.html)

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=919bface228f3431c6602a550de0e3ed

Reply:
It's easier. I pass around arguments where it makes sense but there tend to be places in CLI programs where it's convenient to access the arguments without passing them all the way through nested logic. I've found this pattern works well. It's not essential, though, and I don't think we actually take advantage of it yet.

Reply:
Is there a benefit to using a `with_*` function to provide the options, over accessing the options more directly via the `Mutex`? Or just a style preference?

Reply:
Style preference.